packages/field/src/react/field.tsx

```diff
@@ -0,0 +1,160 @@
+import Halo from '@pluralsight/ps-design-system-halo'
+import {
+  HTMLPropsFor,
+  ValueOf,
+  combineFns
+} from '@pluralsight/ps-design-system-util'
+import Theme, { names as themeNames } from '@pluralsight/ps-design-system-theme'
+import { compose, css } from 'glamor'
+import React, {
+  ForwardRefExoticComponent,
+  MouseEventHandler,
+  RefAttributes,
+  ReactNode,
+  SyntheticEvent,
+  forwardRef,
+  useCallback,
+  useMemo,
+  useRef
+} from 'react'
+
+import { appearances, sizes } from '../vars'
+import stylesheet from '../css/field'
+
+import Input from './input'
+import Label from './label'
+import SubLabel from './sub-label'
+import TextArea from './text-area'
+
+const styles = {
+  container: (opts: { disabled?: boolean; error?: boolean }) =>
+    compose(
+      css(stylesheet['.psds-field__container']),
+      opts.disabled && css(stylesheet['.psds-field__container--disabled']),
+      opts.error && css(stylesheet['.psds-field__container--error'])
+    ),
+  field: (opts: { size?: string }) =>
+    compose(
+      css(stylesheet['.psds-field']),
+      css(stylesheet[`.psds-field--${opts.size}`])
+    ),
+  prefix: () => css(stylesheet['.psds-field__prefix']),
+  suffix: () => css(stylesheet['.psds-field__suffix'])
+}
+
+type InputElements = HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement
+
+interface FieldProps extends Omit<HTMLPropsFor<'div'>, 'prefix' | 'ref'> {
```

Comment:
needed to omit the legacy string ref from `HTMLPropsFor`. i don't think we need to support the legacy ref anymore so I propose we just always do this in the utils package

Reply:
thanks for the 👍  - i'll do that in a new pr